test-suite/tests/integration.rs

```diff
@@ -0,0 +1,73 @@
+use libtest_mimic::Arguments;
+use test_suite::{ERunner, RunnerTrait, TestConfig, TestManager};
+
+use std::{
+    fs,
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4, TcpStream},
+    thread,
+    time::Duration,
+};
+
+// Read and parse the TOML file into a TestConfig
+fn read_test_cases() -> Result<TestConfig, anyhow::Error> {
+    let file_path = format!("{}/tests/tests.toml", env!("CARGO_MANIFEST_DIR"));
```

Comment:
You're hardcoding the path to `tests.toml` completely (as in, `/home/myself/repos/winterjs/test-suite/tests/tests.toml`), since `env!` hardcodes the value the environment variable has at *compile time* into the binary.

While it's true that this code will probably always run right after it's compiled and the executable then discarded, meaning this won't cause a problem, I still think we can do it better. Two options:

* use `include_str!` to include the test cases directly into the binary as a string. This way, it'll be simpler to put the code in other environments (such as in a WASM module).
* use a relative path; cargo always launches programs with `pwd` set to `CARGO_MANIFEST_DIR`, so you can just use `tests.toml` and put the file next to `Cargo.toml` or `tests/tests.toml` if you want to keep it here.

Reply:
Yeah, `include_str!` would be better